litellm/utils.py

```diff
@@ -4803,6 +4805,34 @@ def get_provider_fields(custom_llm_provider: str) -> List[ProviderField]:
         return []
 
 
+def create_proxy_transport_and_mounts():
+    proxies = {
+        key: None if url is None else Proxy(url=url)
+        for key, url in get_environment_proxies().items()
+    }
+
+    sync_proxy_mounts = {}
+    async_proxy_mounts = {}
+
+    # Retrieve NO_PROXY environment variable
+    no_proxy = os.getenv("NO_PROXY", None)
+    no_proxy_urls = no_proxy.split(",") if no_proxy else []
+
+    for key, proxy in proxies.items():
+        if proxy is None:
+            sync_proxy_mounts[key] = httpx.HTTPTransport()
+            async_proxy_mounts[key] = httpx.AsyncHTTPTransport()
+        else:
+            sync_proxy_mounts[key] = httpx.HTTPTransport(proxy=proxy)
+            async_proxy_mounts[key] = httpx.AsyncHTTPTransport(proxy=proxy)
```

Comment:
The problem with this is, that you create Transports here, which are different to the Transport you actually want to use later. With this solution, it would use the proxy Transports for all urls, which must use proxy and the no-proxy Transports for the urls which match the `NO_PROXY` environment variable. Which means, it would never use the transport you assign later with the argument `transport` to the `httpx.AsyncClient`.

Instead you have to do it differently:
* create the transport you actually want to use for all connections (for example `AsyncCustomHTTPTransport`)
* assign this proxy to all urls in `proxies`, where the value is `None`
* for every value different to `None`, create a new transport of the type you want to use (for example `AsyncCustomHTTPTransport`) or maybe copy it, and apply the proxy value to it